pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -140,6 +140,16 @@ func makeHookPod(hook *deployapi.LifecycleHook, deployment *kapi.ReplicationCont
 		restartPolicy = kapi.RestartPolicyOnFailure
 	}
 
+	// Transfer any requested volumes to the hook pod.
+	volumes := []kapi.Volume{}
+	for _, volume := range deployment.Spec.Template.Spec.Volumes {
+		for _, name := range exec.Volumes {
+			if volume.Name == name {
```

Comment:
The scope was expanded to just copy volumes generally. I'm not sure why secrets would be special.